### ISSUE TITLE
Iteration 1

### ISSUE THREAD

**Issue by haaawk** [tursodatabase/libsql-shell-go#1]
This is an umbrella issue for the first development iteration.

Here's an overview of what we're planning to do:
![image](https://user-images.githubusercontent.com/1311149/217858723-59c4d82f-ceca-44dc-baf7-fecb1bbd1671.png)

In short we want to build a library (marked with number 1 on the diagram) that has multiple backends:

- Local SQLite file (number 4)
- Local LibSQL file (number 3)
- Turso in the cloud (number 2)

This library should be able to run sql against those backends and return useful diagnostics and error messages.

There are two usages of of the library:

- headless command-line tool like `shell -e "select * from table" dbfile` (number 7)
- command-line interactive UI (number 5) both with standalone command-line tool (number 6) and integrated into TursoCLI (number 8)

For the first iteration we should focus of minimal end-to-end prototype.
Let's select SQLite backend (number 4) for the first iteration.
From frontend perspective let's first do headless command-line option (number 7) and then stand alone interactive command-line tool (number 6) which requires the work on interactive command-line UI (number 5).
All of this requires creation of the library as well (number 1).

So for the first iteration we ignore TursoCLI integration (number 8), LibSQL backend (number 3) and Turso backend (number 2).

# Minimal result of this iteration
We should at least have:

- A command-line tool called `libsql-shell` that can be run like this `libsql-shell "<sql statement>" <path to db file>`
- It should internally use SQLite to read the given db file, execute given SQL statement on that database, save the changes to the file if there were any and print back SQL statements results to the standard output.
- It should not break on wrong SQL statement or wrong db file but gracefully inform user what's wrong
- There should be tests that prove various SQL statements work and that cases like empty db work as well.

# Target result of this iteration
In addition to the minimal result of this iteration we would also like to have:

- Interactive UI that can be started with `libsql-shell <path to db file>`
- It should have a bar at the bottom when I can type and edit the command before I execute it by pressing enter
- It should have a history of commands
- Basically we would like a feature parity with `turso db shell` so we should be able to do at least everything that's possible in current shell embedded into turso CLI
- It would be good to have some tests that can automatically verify that this works 

# Next iteration

In the next iteration, the goal will be to add Turso backend and to embed the UI into current turso CLI. This would give us two benefits:

- turso CLI users will be able to quickly benefit from all the improvements we do in each iteration
- We will make sure that what we're developing works with both turso CLI and turso backend
